crates/mysten-metrics/src/lib.rs

```diff
@@ -630,6 +635,20 @@ pub async fn metrics(
     }
 }
 
+#[macro_export]
+macro_rules! system_invariant_violation {
+    ($msg:expr) => {
```

Comment:
Should we supply a tag or file:line, and add it as a label to the `system_invariant_violations` metric?